### PR TITLE
Feature/new automate page live spy changes

### PR DESCRIPTION
**Pull request by shahanemahesh**

### CODE REVIEW

File: Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/ObjectBindingHelper.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Ginger.BusinessFlowPages_New
+{
+    public class ObjectBindingHelper
+    {
```
Comment (MeniKadosh1) [inline Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/ObjectBindingHelper.cs]: i don't think this class is needed, just create Dic<Agent,SpyPage>, if you still wants to have it so call the class with more specific name like "AgentLiveSpyMap"

File: Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/MainAddActionsNavigationPage.xaml.cs [hunk 2]
```diff
@@ -124,11 +133,37 @@ private void XNavActLib_Click(object sender, RoutedEventArgs e)
...
+        }
+
+        /// <summary>
+        /// This method is used to get the new LiveSpyPage based on Context and Agent
+        /// </summary>
+        /// <returns></returns>
+        private LiveSpyNavPage GetLiveSpyPage()
+        {
+            LiveSpyNavPage liveSpyNavPage = null;
+            if (mLiveSpyNavPages != null && mLiveSpyNavPages.Count > 0)
```
Comment (MeniKadosh1) [inline Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/MainAddActionsNavigationPage.xaml.cs]: you can keep the unique spy page per Agent but please move all relevent code from main page to "LiveSpyNavPage" page, so you will have "LiveSpyPage" and "LiveSpyNavPage" which will manage the agent-spyPage dic and will show in Frame the "LiveSpyPage"

File: Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/WindowsExplorerNavPage.xaml.cs [hunk 3]
```diff
@@ -32,389 +35,98 @@ namespace Ginger.BusinessFlowsLibNew.AddActionMenu
...
-            return isSearched;
-        }
-
-        private bool RefreshFilteredElements()
-        {
-            if (FilteringCreteriaList.Count != 0)
+            if (!isLoaded)
             {
-                if (CheckedFilteringCreteriaList.Count == 0)
+                ApplicationAgent appAgent = AgentHelper.GetAppAgent(context.BusinessFlow.CurrentActivity, context.Runner, context);
```
Comment (MeniKadosh1) [inline Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/WindowsExplorerNavPage.xaml.cs]: why not to take Agent from Context?

File: Ginger/Ginger/BusinessFlowPages_New/NewAutomatePage.xaml.cs [hunk 4]
```diff
@@ -91,24 +91,43 @@ private void SetUIControls()
...
         }
 
+        /// <summary>
+        /// This is the context changed handler
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void Context_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e != null && e.PropertyName == nameof(Activity))
```
Comment (MeniKadosh1) [inline Ginger/Ginger/BusinessFlowPages_New/NewAutomatePage.xaml.cs]: why we need Context_PropertyChanged in NewAutomatePage? the only item which allowed to change values of Automate Context is NewAutomatePage so why it need to listen to Context changes if it is the one who change it?

File: Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/ObjectBindingHelper.cs (hunk 1 above)
Comment (shahanemahesh) [reply Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/ObjectBindingHelper.cs]: As discussed renamed the class to "AgentMappingPage"

File: Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/MainAddActionsNavigationPage.xaml.cs (hunk 2 above)
Comment (shahanemahesh) [reply Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/MainAddActionsNavigationPage.xaml.cs]: Added new SpyPage

File: Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/WindowsExplorerNavPage.xaml.cs (hunk 3 above)
Comment (shahanemahesh) [reply Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/WindowsExplorerNavPage.xaml.cs]: Done

File: Ginger/Ginger/BusinessFlowPages_New/NewAutomatePage.xaml.cs (hunk 4 above)
Comment (shahanemahesh) [reply Ginger/Ginger/BusinessFlowPages_New/NewAutomatePage.xaml.cs]: As discussed changed the code

File: Ginger/Ginger/BusinessFlowPages_New/NewAutomatePage.xaml.cs [hunk 5]
```diff
@@ -154,21 +154,26 @@ private void SetUIControls()
...
+                SetContextAgent(mContext.Agent, mContext.BusinessFlow.CurrentActivity, mContext.Runner, mContext);
+            }
+        }
+
+        /// <summary>
+        /// This method adds the activitychanged handler
+        /// </summary>
+        private void ActivityChangedHandle()
+        {
+            if (mContext.BusinessFlow.CurrentActivity != null)
```
Comment (MeniKadosh1) [inline Ginger/Ginger/BusinessFlowPages_New/NewAutomatePage.xaml.cs]: If should be - "if (mContext..Activity != null)"

**Pull request merged**